internal/vpn.go

```diff
@@ -0,0 +1,136 @@
+// +build vpn
+
+package internal
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+type ConfigFileReq struct {
+	Name        string   `json:"name"`
+	Value       string   `json:"value"`
+	Unencrypted bool     `json:"unencrypted"`
+	OnChanged   []string `json:"on-changed,omitempty"`
+}
+
+type ConfigCreateRequest struct {
+	Reason string          `json:"reason"`
+	Files  []ConfigFileReq `json:"files"`
+}
+
+// Create a private key and return the derived public key
+func generateKey(privKeyPath string) (string, error) {
+	pkey, err := exec.Command("wg", "genkey").Output()
+	if err != nil {
+		return "", err
+	}
+
+	cmd := exec.Command("wg", "pubkey")
+	stdin, err := cmd.StdinPipe()
+	if err != nil {
+		return "", err
+	}
+
+	go func() {
+		defer stdin.Close()
+		stdin.Write(pkey)
+	}()
+
+	pub, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+	if err := ioutil.WriteFile(privKeyPath, pkey, 0640); err != nil {
```

Comment:
Maybe 0600 instead?

Reply:
indeed. must be copy/pasted from something else